metrics.md

```diff
@@ -0,0 +1,47 @@
+|gluster_volume_total_bytes| gluster_volume_total_bytes{volume="xxx"}  104857600	|
+|gluster_volume_used_bytes| gluster_volume_used_bytes{volume="xxx"}  4857600	|
+|gluster_volume_available_bytes|	gluster_volume_available_bytes{volume="xxx"} 9856780	|
+
+#### Brick Metrics:
+|Gluster Promethus schema| example
+|--------|-----------|
+| gluster_brick_inodes_used |	gluster_brick_inodes_used{volume="xxx", "brick_path"="/mnt/xxx", nodeid="xxxxx"} 100	|
+| gluster_brick_inodes_total |	gluster_brick_inodes_total{volume="xxx", "brick_path"="/mnt/xxx", nodeid="xxxxx"} 200 |	
+| gluster_brick_inodes_available | gluster_brick_inodes_available{volume="xxx", "brick"="/mnt/xxx", nodeid="xxxxx"} 100 |
+| gluster_brick_size_bytes | gluster_brick_size_bytes{volume="xxx", brick_path="xxxx", nodeid="xxx"} 1048576	|
+| gluster_brick_used_bytes|	gluster_brick_used_bytes{volume="xxx", brick_path="xxxx", nodeid="xxx"} 1048	|
+| gluster_brick_available_bytes |	gluster_brick_available_bytes{volume="xxx", brick_path="xxx", nodeid="xxx"} 1047576	|
+| gluster_disk_read_ios | gluster_disk_read_ios{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0
+|   gluster_disk_read_merges | gluster_disk_read_merges{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0
+|   gluster_disk_read_sectors | gluster_disk_read_sectors{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0
+|   gluster_disk_read_ticks | gluster_disk_read_ticks{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0
+|   gluster_disk_write_ios | gluster_disk_write_ios{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0
+|   gluster_disk_write_merges | gluster_disk_write_merges{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0 |
+|   gluster_disk_write_sectors | gluster_disk_write_sectors {brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0 |
+|   gluster_disk_write_ticks | gluster_disk_write_ticks {brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0 |
+|   gluster_disk_inflight | gluster_disk_inflight{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0 |
+|   gluster_disk_total_ticks | gluster_disk_total_ticks{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0 |
+|  gluster_disk_time_inqueue | gluster_disk_time_inqueue{brick_path="xxx",device="xxx",disk="vda",host="xxx",id="xxx",instance="xxx:8080",subvolume="xxx",volume="xxx"} 0 |
+
+#### Gluster process Metrics: 
```

Comment:
Hi John, which metrics will be affected by brick mux?. are you asking about disk metrics or process metrics?.

Reply:
@JohnStrunk I think it is better to include the list of bricks that are loaded in a process rather than just one brick. It is very difficult to get per brick memory usage in the process, so this seems like the best we can do at the moment. What do you think?

Reply:
@pranithk That seems reasonable.
How does that change the schema? the `brick_path=` and `volume=` tags will need to be changed in the metrics. Related to this, are the same bricks always mux-ed into a given process? (assuming not...) If the set of bricks in a process changes, we'll need to make sure a sensible thing happens.

Reply:
@aravindavk How do we get this information from gd2?

Reply:
@aravindavk How do we get the information about which bricks from which volumes are in a particular process.

Reply:
- All gluster related processes information is collected using `ps --no-header -ww -o "pid,pcpu,pmem,rsz,vsz,etimes,comm"  -C "glusterd,glusterd2,glusterfsd,glusterfs"` https://github.com/gluster/gluster-prometheus/blob/master/gluster_exporter/metric_ps.go#L127
- Parse `/proc/<pid>/cmdline` to collect additional details about the processes https://github.com/gluster/gluster-prometheus/blob/master/gluster_exporter/metric_ps.go#L96

With the above approach, metrics related to gluster processes are collected without depending on gluster cli or glusterd.

I think for brickmux this approach will not work.

Reply:
@aravindavk Yeah, I think we need a way to query GD2 and find this information. Shall I raise a github issue for this?

Reply:
brick-mux is not yet supported in GD2. https://github.com/gluster/glusterd2/issues/219

Reply:
@aravindavk Okay. How do you propose we go ahead with this? How easy is it to change schema once we decide it?

Reply:
@aravindavk I agree, showing all brick paths in the same metric will add confusion. May be we can do what volume status does, i.e. for each brick we can show pid of the process in which it resides. If there is a memory increase or leaks then the user can learn that multiple bricks will be impacted. Will that be a good model?

Reply:
we cant show all brick paths in single metrics. can the gauge metric have strings as value. if we think in scale, showing all brick paths as metric looks weird.